wallet/src/ui/app/components/filters-tags/index.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useState, useCallback } from 'react';
+import ReactDOM from 'react-dom';
+import { NavLink } from 'react-router-dom';
+
+import { useMutationObserver } from '_hooks';
+
+import st from './Filters.module.scss';
+
+const ELEMENT_ID = '#sui-apps-filters';
+
+function activeTagsFilter({ isActive }: { isActive: boolean }) {
+    return cl({ [st.active]: isActive }, st.filter);
+}
+
+// TODO: extend this interface to include params and functions for the filter tags
+interface Props {
+    name: string;
+    link: string;
+}
+
+type Tags = {
+    tags: Props[];
+};
+
+function FiltersPortal({ tags }: Tags) {
+    const [ready, setReady] = useState(false);
+    const content = document.querySelector(ELEMENT_ID) as HTMLElement;
+
+    const handleMutations = useCallback(
+        (mutations: { type: MutationRecordType }[]) => {
+            mutations.forEach(({ type }: { type: MutationRecordType }) => {
+                if (type === 'childList') setReady(true);
+            });
+        },
+        []
+    );
+
+    useMutationObserver({
+        target: content,
+        options: { childList: true, subtree: true },
```

Comment:
Options needs to be memoized otherwise it'll re-observe the element every single render.

Reply:
Alternatively, change the `useMutationObserver` API to just remove options and hardcode these options there.

Reply:
Ok